File: 283-move-zeroes/283-move-zeroes.php
class Solution {

    /**
     * @param Integer[] $nums
     * @return NULL
     */
    function moveZeroes(&$nums) {
        $zeroArray =[];
        $numArray = [];
        
        foreach($nums as $num){
            if($num == 0){
                $zeroArray[] = 0;
            } else{
                $numArray[] = $num;
            }
        }
        $nums = array_merge($numArray, $zeroArray);
    }
}

File: 283-move-zeroes/283-move-zeroes.kt
class Solution {
    private fun swap (first: Int, second: Int, nums: IntArray) {
        val temp = nums[first]
        nums[first] = nums[second]
        nums[second] = temp
    }
    
    fun moveZeroes(nums: IntArray): Unit {
        var first = 0
        var second = 0
        
        val numsSize = nums.size
        
        while (first < numsSize) {
            if (nums[first] != 0) {
                swap(first, second, nums)
                second++
            }
            
            first++
        }
    }

}